lxd/containers.go

```diff
@@ -1719,6 +1737,16 @@ func (c *lxdContainer) setupMacAddresses(d *Daemon) error {
 			}
 		}
 
+		/*
+		 * propagate the name of the device down for nics if the user
+		 * didn't specify one. This is probably what they wanted
+		 * anyway, and lets us match against network names once they go
+		 * into lxc, e.g. to pull out the veth name for `lxc info`.
+		 */
+		if _, ok := d["name"]; !ok {
+			d["name"] = name
+		}
+
```

Comment:
If "name" changes from being unset to being set to "lxcbr0", it will mean that the network device inside the container will change from being called "eth0" to being called "lxcbr0".

As "eth0" is hardcoded in quite a few places, most importantly, /etc/network/interfaces, the container will then boot without networking.